src/expr/src/scalar/func.rs

```diff
@@ -1841,23 +1841,21 @@ where
     }
 }
 
-fn parse_timezone<'a>(a: Datum<'a>) -> Result<Timezone, EvalError> {
-    let tz = a.unwrap_str();
+pub(crate) fn parse_timezone<'a>(tz: &'a str) -> Result<Timezone, EvalError> {
     tz.parse()
         .map_err(|_| EvalError::InvalidTimezone(tz.to_owned()))
 }
 
-fn timezone_time<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
-    let offset = match parse_timezone(a)? {
+fn timezone_time<'a>(tz: Timezone, t: NaiveTime) -> Datum<'a> {
```

Comment:
and here you can just use the anonymous lifetime:

```suggestion
fn timezone_time(tz: Timezone, t: NaiveTime) -> Datum<'_> {
```